### PR TITLE
Add integration test for #36430

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY

https://github.com/ansible/ansible/pull/36430

##### ISSUE TYPE

 - Test Pull Request

##### ANSIBLE VERSION

```
devel
```

### CODE REVIEW

File: test/integration/targets/loops/tasks/main.yml [hunk 1]
```diff
@@ -176,3 +176,29 @@
...
+    loopthis:
+        - name: foo
+          label: foo_label
+        - name: bar
+          label: bar_label
+
+- name: check that item label is updated each iteration
+  debug:
+    msg: "{{ looped_var.name }}"
+  with_items: "{{ loopthis }}"
```
Comment (webknjaz) [inline test/integration/targets/loops/tasks/main.yml:195]: How about `with_dict`?

Comment (mkrizek) [reply test/integration/targets/loops/tasks/main.yml:195]: `loopthis` is a `list`, I don't see how would changing it to `dict` and using `with_dict` make testing the loop label better?

File: test/integration/targets/loops/tasks/main.yml [hunk 2]
```diff
@@ -176,3 +176,29 @@
...
+  loop_control:
+    loop_var: looped_var
+    label: "looped_var {{ looped_var.label }}"
+  register: output
+
+- assert:
+    that:
+      - "output.results[0]['_ansible_item_label'] == 'looped_var foo_label'"
+      - "output.results[1]['_ansible_item_label'] == 'looped_var bar_label'"
```
Comment (jctanner) [inline test/integration/targets/loops/tasks/main.yml:204]: _ansible_item_label is gauranteed to always be in the looped results?

Comment (mkrizek) [reply test/integration/targets/loops/tasks/main.yml:204]: Who knows :) But as I understood it, if there's loop and label defined in that loop, then yes: https://github.com/ansible/ansible/blob/devel/lib/ansible/executor/task_executor.py#L334

Comment (bcoca) [reply test/integration/targets/loops/tasks/main.yml:204]: yes, even if user does not define it, you get the 'default label value'  `{{ item }}`

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/55389/36/tests

**Pull request merged**